### PR TITLE
Alter tests to reuse cluster components

### PR DESCRIPTION
**Pull request by GuptaManan100**
## Description
This PR alters the vtorc tests to reuse the cluster components to speed them up. It also adds an additional test that checks for the failure case of circular replication. i.e. Primary and replica are replicating from each other.

Before the PR
BeforeChange

After the changes
AfterChange

The test time goes from almost 7 minutes to almost 4 minutes which is a 42.8% decrease in runtime.

## Related Issue(s)
 - #6769

## Checklist
- [x] Tests were added or are not required
- [x] Documentation was added or is not required

### CODE REVIEW

File: go/test/endtoend/cluster/vtorc_process.go [hunk 1]
```diff
@@ -60,7 +61,7 @@ func (orc *VtorcProcess) Setup() (err error) {
 	orc.proc.Args = append(orc.proc.Args, orc.ExtraArgs...)
 	orc.proc.Args = append(orc.proc.Args, "-alsologtostderr", "http")
 
-	errFile, _ := os.Create(path.Join(orc.LogDir, "orc-stderr.txt"))
+	errFile, _ := os.Create(path.Join(orc.LogDir, "orc-stderr.txt"+time.Now().String()))
```
Comment (shlomi-noach) [inline go/test/endtoend/cluster/vtorc_process.go:64]: - nit: keep `.txt` as file extension. Inject the timestamp before the extension.
- nit: `time.Now().String()` has spaces. Perhaps use unix time converted to string, which is only numeric.

File: go/test/endtoend/cluster/vttablet_process.go [hunk 2]
```diff
@@ -231,7 +245,27 @@ func (vttablet *VttabletProcess) WaitForTabletTypesForTimeout(expectedTypes []st
 		}
 	}
 	return fmt.Errorf("Vttablet %s, current status = %s, expected status [%s] not reached, details: %v",
-		vttablet.TabletPath, status, strings.Join(expectedTypes, ","), vttablet.GetStatusDetails())
+		vttablet.TabletPath, status, strings.Join(expectedStatuses, ","), vttablet.GetStatusDetails())
+}
+
+// WaitForTabletTypesForTimeout waits till the tablet reaches to any of the provided types
+func (vttablet *VttabletProcess) WaitForTabletTypesForTimeout(expectedTypes []string, timeout time.Duration) error {
+	timeToWait := time.Now().Add(timeout)
```
Comment (shlomi-noach) [inline go/test/endtoend/cluster/vttablet_process.go:253]: nit: `timeToWait` is a confusing name. Perhaps rename to `waitUntil` or `expierTime` or similar?

File: go/vt/vtctl/vtctl.go [hunk 3]
```diff
@@ -798,6 +801,26 @@ func commandDeleteTablet(ctx context.Context, wr *wrangler.Wrangler, subFlags *f
...
+		return err
+	}
+	for _, tabletAlias := range tabletAliases {
+		if err := wr.DemoteMasterTablet(ctx, tabletAlias); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
```
Comment (shlomi-noach) [inline go/vt/vtctl/vtctl.go:823]: I assume this new command is only used for testing? But this still exposes the command to everyone and can confuse people.

**Review by shlomi-noach — COMMENTED** [review]
Good work! Some nits inside

File: go/vt/vtctl/vtctl.go (hunk 3 above)
Comment (GuptaManan100) [reply go/vt/vtctl/vtctl.go:823]: Yes, we had a discussion regarding this and have decided to remove this function. We will create a separate function which demotes a primary, which isn't exposed for end users.

File: go/test/endtoend/cluster/vtorc_process.go (hunk 1 above)
Comment (GuptaManan100) [reply go/test/endtoend/cluster/vtorc_process.go]: Changed

File: go/test/endtoend/cluster/vttablet_process.go (hunk 2 above)
Comment (GuptaManan100) [reply go/test/endtoend/cluster/vttablet_process.go]: Changed

File: go/vt/vtctl/vtctl.go (hunk 3 above)
Comment (GuptaManan100) [reply go/vt/vtctl/vtctl.go]: Moved this functionality to a local function in the testing environment

**Review by shlomi-noach — APPROVED** [review]
Nice work! :rocket:

File: go/vt/wrangler/tablet.go [hunk 4]
```diff
@@ -151,7 +151,7 @@ func (wr *Wrangler) DeleteTablet(ctx context.Context, tabletAlias *topodatapb.Ta
 		return err
 	}
 
-	return nil
+	return
```
Comment (deepthi) [inline go/vt/wrangler/tablet.go:154]: nit: this seems unnecessary.

**Review by deepthi — APPROVED** [review]
Love the reduction in test run time :)

File: go/vt/wrangler/tablet.go (hunk 4 above)
Comment (GuptaManan100) [reply go/vt/wrangler/tablet.go]: Fixed

**Pull request merged**